server/etcdserver/cindex/cindex.go

```diff
@@ -85,11 +82,14 @@ func (ci *consistentIndex) SetConsistentIndex(v uint64) {
 }
 
 func (ci *consistentIndex) UnsafeSave(tx backend.BatchTx) {
-	bs := ci.bytesBuf8
-	binary.BigEndian.PutUint64(bs, ci.consistentIndex)
-	// put the index into the underlying backend
-	// tx has been locked in TxnBegin, so there is no need to lock it again
-	tx.UnsafePut(metaBucketName, consistentIndexKeyName, bs)
+	index := atomic.LoadUint64(&ci.consistentIndex)
+	if index > 0 {
```

Comment:
You may flip this condition and fail fast
```
if index > 0 {
    return
}
...
```

Reply:
Done